### DISCUSSION TITLE
Question about the AD of jax.scipy.linalg.eigh

### DISCUSSION THREAD

**Discussion by DoTulip** [jax-ml/jax#8148] [Q&A] [answered]
Hello, JAX team @jakevdp @shoyer. In scientific computing, the generalized eigenvalue problem is a type of problem that researchers often encounter. In Scipy, we can use scipy.linalg.eigh to solve the generalized eigenvalue problem, but I noticed that currently we cannot solve this problem by using jax.scipy.linalg.eigh. Here, I have some questions and look forward to your reply.
1. Will JAX support solving the generalized eigenvalue problems in the future? 
2. For some specific research fields, such as structural dynamics analysis. We need to solve the generalized eigenvalue problem as following: **Ku=wMu**, where both **K** and **M** are symmetric positive definite matrices. Is there a straightforward way for us to provide eigenvalue **w** derivatives？

I am looking forward to your reply, thank you.

**Comment by shoyer** [comment, marked as answer]
There are no immediate plans (that I know of) to implement generalized eigenvalue solvers into JAX, but certainly contributions in this area would be very welcome. If you're interested in helping, I would suggest opening an issue to discuss.
